src/plugins/customSounds/index.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2023 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { DataStore } from "@api/index";
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { React } from "@webpack/common";
+
+import { SoundOverrideComponent } from "./components/SoundOverrideComponent";
+import { makeEmptyOverride, SoundOverride, soundTypes } from "./types";
+
+const OVERRIDES_KEY = "CustomSounds_overrides";
+let overrides: Record<string, SoundOverride> = soundTypes.reduce((result, sound) => ({ ...result, [sound.id]: makeEmptyOverride() }), {});
+
+const settings = definePluginSettings({
+    overrides: {
+        type: OptionType.COMPONENT,
+        description: "",
+        component: () =>
+            <>{soundTypes.map(type =>
```

Comment:
needs some room to breathe
```suggestion
            <>
            	{soundTypes.map(type =>
```